### DISCUSSION TITLE
Head doesnt work?

### DISCUSSION THREAD

**Discussion by bradennapier** [react-static/react-static#1535] [Q&A]
When I try to use the `Head` from react-static it ends up with duplicate head meta tags -- and it uses the original head tags ..

so how do i set it up so that different pages have different head meta when shared?   isnt this like... one of ht emain reasons to have static lol

When running locally it works fine - i only have one set of meta tags... but once it is on the server i end up with duplicates.

The original meta tags used do not have the `data-react-helmet` prop ... so not sure what react-static is doing here... clearly the `Head` used with `Document` an the head used when importing from react-static are completely different?

Do we have to keep the meta for every page and render in the `Document` ?  This should be mentioned if so and is quite annoying since it specifically says you can just call `Head` anywhere which doesn't appear to be true.

**Comment by SleeplessByte** [comment]
>  and it uses the original head tags

How is it rendering a head without the `Head` tag?

> This should be mentioned if so and is quite annoying since it specifically says you can just call Head anywhere which doesn't appear to be true.

It's using `react-helmet`, I would defer to that.

**Reply by bradennapier** [reply]
The `Document` in the config gets a `Head` given to it.  If I use the `Head` from `react-static` here it doesnt work at all.  

If I use the `Head` from `react-static` in other components, it doesn't render on the server side and is duplicated so that the head tags from the `Document` in config as well as the tags added from the imported `Head` are both showing.

**Comment by SleeplessByte** [comment]
You can find the default document [here](https://github.com/react-static/react-static/blob/18191c80a8d31debda5106edf468e7daa4ac517e/packages/react-static/src/static/components/RootComponents.js).

I don't quite understand what you're doing, or how you're doing it, but in general if you provide your _own_ `Document`, you'd use

```tsx
{
  Document: ({
    Html,
    Head,
    Body,
    children,
    state: { siteData, routeInfo },
  }) => {
    return (
    <Html lang="en-US">
      <Head
          defaultTitle="Example.org - The default site title if none is given"
          titleTemplate="Example.org - %s"
      >
        <html lang="en" />
        <meta name="description" content="some description" />
        <meta name="keywords" content="my, great, keywords, " />

        <script id="google-tag-manager">{`
        (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
        new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
        j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
        'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
        })(window,document,'script','dataLayer','GTM-XXX');
        `}</script>
      </Head>
      <Body>{children}</Body>
    </Html>
  )}
}
```

On pages, or wherever, importing `Head` from `react-static` allows you to inject / override tags. For example, this is taken directly from one of our projects:

```tsx
 <ProductPage data-page="product">
  <Head title={`${name} in ${city.name}, ${state.name} - ${category.name}`}>

    {<link rel="canonical" href={seoCanonicalHref} />}

    <meta name="description" content={metaDescription} />
    <meta name="keywords" content={metaKeywords} />
    <script type="application/ld+json">{
      JSON.stringify({
        "@context": "https://schema.org",
        "@type": "MedicalOrganization",
          "mainEntityOfPage": {
            "@type": "WebPage",
            "@id": `https://example.org${rootProductPath(category.slug, state.slug, city.slug, slug)}`
          },
          "name": name,
          "address": address,
          "image": [
          ],
          "medicalSpecialty": "Nursing",
          "description": seoDescription
        }, undefined, 2)
      }
    </script>

  </Head>
      ...
</ProductPage>
```

This combines the `Head` and overrides any duplicates. If you see different behaviour, you either are not using a compatible version of react-helmet, or are using a different technique to update the Head (e.g. via a package).

**Reply by bradennapier** [reply]
That is exactly how we are doing it.   Only thing that i can think of is that its being broken by the filesystem plugin since we don't use normal routes.

I have changed to just dynamically setting the meta tags for each route when it renders the `Document` which works for now at least.

By default the hot reload actually doesn't work well either - so we are also using the `hot-loader` patched version of react-dom which i guess would be the cause as well but seems less likely

**Reply by SleeplessByte** [reply]
It feels like there might be more dependency mismatches. Just like with create-react-app and expo, bumping everything to the latest versions won't work as expected.

Regardless; I'm happy you have a way around it. Because of the current pandemic we don't have the capacity to go on this quest of trying to figure out what is actually breaking, but if there is a way where this behaviour can be reproduced with a blank project (and then only adding, for example, the file system plugin), would help tremendously for anyone trying to resolve the issues you describe.

**Comment by mhluska** [comment]
I'm seeing the same thing on react-static v7.5.3. Works fine locally but produces duplicates after building.

Submitted a bug report here: https://github.com/react-static/react-static/issues/1630
